### ISSUE TITLE
Does OT support a boolean geometry tool?

### ISSUE THREAD

**Issue by ForestEco** [opentoonz/opentoonz#5583]
### Ask a question related to OpenToonz

Basically seeing if there's a tool that would take shapes and cut them out of other objects, intersect, so on like they do in blender?

**Comment by RodneyBaker** [comment]
No but...

There are some ways we can merge and cut out shapes.
I'll describe one way here...

Draw two (or more) shapes with Geometry Tool (or however you wish) and either copy/paste them together or merge.
If using black and white only we can use black as the shape that will be left and white as the cutaway parts.
In this way we get a primative boolean (on/off) means of creating a shape.

When satisfied with the shape, convert this raster image to vector using Outline mode and if the lines aren't connected appropriately use 'Simply Vectors' to join them.

It can be useful to convert from vector to toonz raster and then to raster to refine the image prior to the final step of 'merging' the shape and there are some things to watch out for.

Here's a rather random example of black and white shapes used to create a vector shape:

![image](https://github.com/user-attachments/assets/3d121baa-4855-4c2b-9e3e-9671fe8e420a)


Intersects are mostly a matter of just coloring the appropriate area white (or black) so that it will cut accordingly.


So, in other words, you might draw your shapes/objects in either raster, toonz raster or vector (vector if you want to edit/tweak the shapes in the process) with the goal of copy/pasting or merging down to raster.  Then with the raster we can convert to vector to get the final shape. (It can help to binarize the raster just before conversion to guarantee black and white.  This doesn't mean other colors can be used but if we only have two colors it's easier to predict the result.

There's a lot more we can do... Just grouping shapes can create merges and intersects.

Perhaps you can share an example of what you might wish to use a boolean cutter on and we can explore that.

**Comment by RodneyBaker** [comment]
Here's a quick attempt at a slice of cheese:

![image](https://github.com/user-attachments/assets/8201b960-ef8a-4027-9e26-993225a5452f)

**Comment by ForestEco** [comment]
Thanks for the description, I will give it a try. I was just trying to make a very small screw shape, the threads need little gaps, so I ended up drawing lines out, but to really make it realistic, you'd want to remove some of the surrounding mass between the threads. How would you animate a screw so it looks like it has motion? I tried shearing the angle and squashing to stretch/exaggerate.

**Comment by ForestEco** [comment]
I am not sure if I have learned how to merge the column levels yet.

**Comment by RodneyBaker** [comment]
While I'm not entirely sure this would be my ideal approach the first thought that comes to mind for a 'screw type motion' I'd consider saving a vector drawing of screw threads as a custom vector brush and then... draw using that brush... or select a draw stroke or strokes and apply that custom style to it.  Getting the screw threads to move might be as simple as deleting a little from the end of the vector stroke which in turn will make the screw threads look like it is moving.

I'll try to put together a project file a little later this evening but... you might consider joining one of the Opentoonz discords where most who would respond here will share information there.  I'm not opposed to posting the information and files here but there's an inherent delay in back and forth responses.  I post the links from time to time so if you are interested and can't find a link just let me know.  If not, we'll try to press on here.
